fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/ExplainTokens.java

```diff
@@ -0,0 +1,1110 @@
+    public ExplainTokens addKeyword(@Nonnull final String keyword) {
+        return addKeyword(Token.DEFAULT_EXPLAIN_LEVEL, keyword);
+    }
+
+    @Nonnull
+    public ExplainTokens addKeyword(final int explainLevel, @Nonnull final String keyword) {
+        add(new KeywordToken(explainLevel, keyword));
+        return this;
+    }
+
+    @Nonnull
+    public ExplainTokens addComma() {
+        return addComma(Token.DEFAULT_EXPLAIN_LEVEL);
+    }
+
+    @Nonnull
+    public ExplainTokens addComma(final int explainLevel) {
+        add(new CommaLikeToken(explainLevel, ","));
+        return this;
+    }
+
+    @Nonnull
+    public ExplainTokens addCommaAndWhiteSpace() {
+        return addCommaAndWhiteSpace(Token.DEFAULT_EXPLAIN_LEVEL);
+    }
+
+    @Nonnull
+    public ExplainTokens addCommaAndWhiteSpace(final int explainLevel) {
+        addComma(explainLevel).addWhitespace(explainLevel);
+        return this;
+    }
+
+    @Nonnull
+    public ExplainTokens addAliasDefinition(@Nonnull final CorrelationIdentifier alias) {
+        return addAliasDefinition(Token.DEFAULT_EXPLAIN_LEVEL, alias);
+    }
+
+    @Nonnull
+    public ExplainTokens addAliasDefinition(final int explainLevel,
+                                            @Nonnull final CorrelationIdentifier alias) {
+        add(new AliasDefinitionToken(explainLevel, alias));
+        return this;
+    }
+
+    @Nonnull
+    public ExplainTokens addCurrentAliasDefinition(@Nonnull final CorrelationIdentifier alias) {
+        return addCurrentAliasDefinition(Token.DEFAULT_EXPLAIN_LEVEL, alias);
+    }
+
+    @Nonnull
+    public ExplainTokens addCurrentAliasDefinition(final int explainLevel,
+                                                   @Nonnull final CorrelationIdentifier alias) {
+        add(new CurrentAliasDefinitionToken(explainLevel, alias));
+        return this;
+    }
+
+    @Nonnull
+    public ExplainTokens addAliasReference(@Nonnull final CorrelationIdentifier alias) {
+        return addAliasReference(Token.DEFAULT_EXPLAIN_LEVEL, alias);
+    }
+
+    @Nonnull
+    public ExplainTokens addAliasReference(final int explainLevel,
+                                           @Nonnull final CorrelationIdentifier alias) {
+        add(new AliasReferenceToken(explainLevel, alias));
+        return this;
+    }
+
+    @Nonnull
+    public ExplainTokens addOpeningParen() {
+        return addOpeningParen(Token.DEFAULT_EXPLAIN_LEVEL);
+    }
+
+    @Nonnull
+    public ExplainTokens addOpeningParen(final int explainLevel) {
+        add(new BracketsToken(explainLevel, true, "("));
+        return this;
+    }
+
+    @Nonnull
+    public ExplainTokens addClosingParen() {
+        return addClosingParen(Token.DEFAULT_EXPLAIN_LEVEL);
+    }
+
+    @Nonnull
+    public ExplainTokens addClosingParen(final int explainLevel) {
+        add(new BracketsToken(explainLevel, false, ")"));
+        return this;
+    }
+
+    @Nonnull
+    public ExplainTokens addOpeningBracket() {
```

Comment:
In US English, a bracket is one of these: `[]`, whereas in UK English, a bracket is one of these: `()`. Our style is to conform with US English, but it might still be less ambiguous to call these `SquareBracket`s.

Reply:
done.